.github/workflows/ci.yaml

```diff
@@ -19,7 +19,7 @@ jobs:
 
     strategy:
       matrix:
-        os: [ubuntu-20.04]
+        os: [ubuntu-21.04]
```

Comment:
This doesn't exist.

See these links:

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources
https://github.com/actions/virtual-environments/tree/main/images/linux

This means we must get a suitable version of Qt from somewhere else.

The simplest answer is to also use vcpkg for this build, but there is one big issue: our GitHub Actions dependency cache is almost full.
This means either the Linux build would have to be entirely uncached (i.e., dependencies built from scratch every time), or maybe we could just build and cache Qt and use the system provided versions of the rest of the dependencies.
This is easy to do, but it means that the builds produced by the CI won't be usable on other machine unless the user tries to precisely match the linked Qt version which defeats the purpose anyway.

Alternatively, we could switch the Ubuntu CI to entirely static builds. The builds would still have to be entirely uncached of course, but at least the builds produced would be usable.

Another possible solution: to get around the GitHub Actions available cache space limitation of 5 GiB per repo, we could create another repo (or as many as necessary) solely for the purpose of building and publishing caches for the main repo to use. Kinda inconvenient, but it should work just fine. I wish GitHub would provide more caching space...

---

Even so, there is still yet another problem: the CI will no longer be testing for the exact environment we want it to test (stock Ubuntu 21.04), so it becomes less useful. To really get around this, I see 2 options:

- Wait for GitHub Actions to provide a 21.04 runner. This may never actually happen, as they will probably only publish runners based on LTS versions.
- Create a 21.04 chroot and build inside it. Probably a bit tricky, but should be doable. The good thing about this solution is that there would be no caching concerns. The build time would increase a bit due to additional step of constructing the chroot, but probably not by much, since at the end of the day, it's mostly just a bunch of extra package installations with `apt`. Another benefit is that the chroot configuration would also be useful to upgrade to future non-LTS releases further down the line.

Reply:
CI fails us again.

Another problem is libtorrent 2, which we planned to switch to in the 4.4 release.

Reply:
@glassez 

> CI fails us again.

Of course, when you upgrade requirements of the application, sometimes you may also have to upgrade some quite fundamental things in the CI environment. This is true no matter the CI vendor. I would not consider this a failure, it's just how it is. The chroot solution for the Ubuntu CI seems the most promising to me, I'll investigate it.

> Another problem is libtorrent 2, which we planned to switch to in the 4.4 release.

I'm working on bringing libtorrent 2 to vcpkg: https://github.com/microsoft/vcpkg/issues/17694, feel free to help with such efforts. Also, 1.2.14 will be released pretty soon, which I will also port there.

Reply:
>Of course, when you upgrade requirements of the application, sometimes you may also have to upgrade some quite fundamental things in the CI environment. This is true no matter the CI vendor. I would not consider this a failure, it's just how it is.

Yes, but it still looks too burdensome. It causes more worries than even upgrading the app itself. This doesn't look acceptable to me. Apparently, we should use some kind of environment configuration that is less sensitive to such changes. We are still waiting for the transition to Qt 6. And maybe first on Qt 5.15.2 as the last step on the way there.
>The chroot solution for the Ubuntu CI seems the most promising to me,

At least it won't require building Qt from source.
> I'll investigate it.

👍

Reply:
@glassez 

> Yes, but it still looks too burdensome. It causes more worries than even upgrading the app itself. This doesn't look acceptable to me. Apparently, we should use some kind of environment configuration that is less sensitive to such changes. We are still waiting for the transition to Qt 6. And maybe first on Qt 5.15.2 as the last step on the way there.

This is a more of a problem with Ubuntu itself than with any other thing. Ubuntu won't have Qt6 in the repos until at least 21.10. But I would expect the situation to improve once the transition to Qt6 is complete. If not, maybe we need to re-evaluate our distribution policy (e.g. sunset the PPA and "official Ubuntu OOTB support" and transition to AppImage, for example).

Also, Ubuntu 21.04 has Qt 5.15.2, so you could just bump Qt to that version. https://packages.ubuntu.com/hirsute/qtbase5-dev

Reply:
>Also, Ubuntu 21.04 has Qt 5.15.2, so you could just bump Qt to that version already.

@sledgehammer999, @Chocobo1
Do we really require this intermediate step (Qt 5.14) before raise Qt to 5.15.2?

Reply:
>I'm working on bringing libtorrent 2 to vcpkg: microsoft/vcpkg#17694

It doesn't look like there's any robust activity going on there... The last (and only) answer was given 21 days ago. BTW, is there something wrong with it?

Reply:
@glassez 

> It doesn't look like there's any robust activity going on there... The last (and only) answer was given 21 days ago. BTW, is there something wrong with it?

I know what I need to know, I'm working locally on it. Once it's ready I'll link the PR, no worries.